### PR TITLE
Improve repository documentation

### PR DESCRIPTION
**Pull request by kenoir**
## What does this change?

This change updates the README at the root of this project to contain a better description of the overall purpose of the service catalogue and adds further READMEs for each project with more context for each.

[Preview Link](https://github.com/guardian/github-lens/blob/rk/improve-readmes/README.md)

## Why?

So folk can understand what this repository is for and work with it more easily.

### CODE REVIEW

File: packages/common/src/expressRoutes.ts [hunk 1]
```diff
@@ -32,11 +32,18 @@ export const getDescribeRouterHandler = (router: Router) => {
...
 						break;
 					case '/repos/:name':
 						info = 'Show repo with the provided name, if it exists';
 						break;
+					case '/teams':
+						info =
+							'Show all teams, with the repositories they own';
+						break;
+					case '/teams/:name':
+						info = 'Show team with the provided name, if it exists';
```
Comment (JuliaBrigitte) [inline packages/common/src/expressRoutes.ts:45]: Should be info = 'Show team and the repositories it owns, if team exists'

File: packages/common/src/expressRoutes.ts [hunk 2]
```diff
@@ -32,11 +32,18 @@ export const getDescribeRouterHandler = (router: Router) => {
 						break;
 					case '/repos':
 						info =
-							'Show all repos, or when ?name=searchString is given search for searchString';
+							'Show all repos, with their team owners';
```
[the remark this answers is not in the document]
Comment (NovemberTang) [reply packages/common/src/expressRoutes.ts:35]: i think if we keep this we should elaborate a little bit as i'm not sure what searchString means. What am I able to search for using this?

[the remark this answers is not in the document]
Comment (kenoir) [reply packages/common/src/expressRoutes.ts:35]: I don't think the search function works anywhere at the moment (in the `github-lens` api it's hardcoded to return results for "riff"!). See: https://github.com/guardian/github-lens/blob/main/packages/github-lens-api/src/app.ts#L37

We should have a working implementation before we describe it here.

**Review by NovemberTang — APPROVED** [review]

File: packages/common/src/expressRoutes.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (JuliaBrigitte) [reply packages/common/src/expressRoutes.ts]: It did definitely worked for me, when I checked it in, will check.

**Pull request merged**